tests/unit_tests/blas/level1/iamax.cpp

```diff
@@ -129,35 +130,77 @@ int test(device* dev, oneapi::math::layout layout, int N, int incx) {
 class IamaxTests
         : public ::testing::TestWithParam<std::tuple<sycl::device*, oneapi::math::layout>> {};
 
-TEST_P(IamaxTests, RealSinglePrecision) {
-    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, 2));
-    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, 1));
-    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, -3));
+TEST_P(IamaxTests, RealSinglePrecisionBaseZero) {
+    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, 2,
+                                  oneapi::math::index_base::zero));
+    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, 1,
+                                  oneapi::math::index_base::zero));
+    EXPECT_TRUEORSKIP(test<float>(std::get<0>(GetParam()), std::get<1>(GetParam()), 1357, -3,
+                                  oneapi::math::index_base::zero));
 }
-TEST_P(IamaxTests, RealDoublePrecision) {
+TEST_P(IamaxTests, RealSinglePrecisionBaseOne) {
```

Comment:
I don't think we need to duplicate all the tests to test the parameter `index_base`. If the same logic were applied to other parameters and operations the number of tests would not be manageable. I would suggest to only have `RealSinglePrecision` and one call to the test function with `index_base::one`. Same for other precisions and for iamin.

Reply:
fixed in 10760de

Reply:
I think it would be more consistent with the rest of the tests in the blas domain if the tests in `RealSinglePrecisionBaseOne` were merged in `RealSinglePrecision` (and only one of the sizes with an index_base one would be enough IMO).

Reply:
Hi @Rbiessy, in 4c21de8, I still kept 2 cases for index_base::one. One's inc is positive (normal case), and another's is negative (we need to return zero. There is no need to +1 or -1).